lib/system/dollars.nim

```diff
@@ -111,7 +116,7 @@ proc collectionToString[T](x: T, prefix, separator, suffix: string): string =
     else:
       result.add(separator)
 
-    when value isnot string and value isnot seq and compiles(value.isNil):
```

Comment:
Old version is preferable to me.

Reply:
Particularly in this pr. It is is actually important to not us ``isNil``, because ``isNil`` doesn't exist for distinct types.